pkgs/applications/misc/dmenu-rs/default.nix

```diff
@@ -0,0 +1,113 @@
+{ lib
+, fetchFromGitHub
+, rustPlatform
+, clang
+, libclang
+, llvm
+, m4
+, pkg-config
+, expat
+, fontconfig
+, libX11
+, libXft
+, libXinerama
+, python3
+}:
+
+# The dmenu-rs package has extensive plugin support. However, this derivation
+# only builds the package with the default set of plugins. If you'd like to
+# further customize dmenu-rs you can build it from the source.
+# See: https://github.com/Shizcow/dmenu-rs#plugins
+rustPlatform.buildRustPackage rec {
+  pname = "dmenu-rs";
+  version = "5.5.1";
+
+  src = fetchFromGitHub {
+    owner = "Shizcow";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-WpDqBjIZ5ESnoRtWZmvm+gNTLKqxL4IibRVCj0yRIFM=";
+  };
+
+  nativeBuildInputs = [
+    clang
```

Comment:
with bindgenHook you presumably need neither libclang, llvm nor clang.

Reply:
I was able to drop `llvm` and `libclang`, but it wouldn't build without keeping `clang`.

Reply:
please drop clang and modify https://github.com/Shizcow/dmenu-rs/blob/master/config.mk#L14 as suggested by the error message

Reply:
Are you suggesting I modify this upstream at `Shizcow/dmenu-rs` or through a patch in the nix derivation?

Reply:
downstream is ok (either as a patch or as `substituteInplace config.mk --replace 'clang' '$(CC)'` in postPatch for example)
But hard coding clang as compiler is not distro-friendly in general so you may submit this change upstream if so you wish, after we merge this PR. Using $(CC) is more portable and preferable.

Reply:
I've added it as a patch here, but also opened [this PR upstream](https://github.com/Shizcow/dmenu-rs/pull/44).

Reply:
I think you forgot to remove the dependency on clang. Thank you for reaching upstream :)

Reply:
Oh you're right, woops

Reply:
Removed